src/addrman.cpp

```diff
@@ -562,10 +562,11 @@ bool AddrManImpl::AddSingle(const CAddress& addr, const CNetAddr& source, int64_
 
     if (pinfo) {
         // periodically update nTime
-        bool fCurrentlyOnline = (GetAdjustedTime() - addr.nTime < 24 * 60 * 60);
-        int64_t nUpdateInterval = (fCurrentlyOnline ? 60 * 60 : 24 * 60 * 60);
-        if (addr.nTime && (!pinfo->nTime || pinfo->nTime < addr.nTime - nUpdateInterval - nTimePenalty))
+        const bool currently_online{std::chrono::seconds{GetAdjustedTime()} - std::chrono::seconds{addr.nTime} < 24h};
```

Comment:
nit: could add include for` <chrono>` to addrman

Reply:
This commit requires `using namespace std::chrono_literals;`, so I've `#include <util/time.h>` instead.

Reply:
Removed commit for now